File: library/hlscache/src/main/java/com/kingz/mobile/libhlscache/utils/Tools.java
package com.kingz.mobile.libhlscache.utils;

/**
 * Created 2017/11/21.
 */
public class Tools {

    public static String getAbsoluteUrl(String baseUrl, String rawUrl) {
        if (rawUrl.startsWith("http")) {
            return rawUrl;
        } else {
            return baseUrl.substring(0, baseUrl.lastIndexOf('/')) + rawUrl;
        }
    }

}

File: library/hlscache/src/main/java/com/kingz/mobile/libhlscache/utils/Tools.kt
package com.kingz.mobile.libhlscache.utils

object Tools {
    @JvmStatic
    fun getAbsoluteUrl(baseUrl: String, rawUrl: String?): String {
        return if (rawUrl?.startsWith("http") == true) {
            rawUrl
        } else {
            baseUrl.substring(0, baseUrl.lastIndexOf('/')) + rawUrl
        }
    }
}